src/test/pybind/test_rbd.py

```diff
@@ -2165,10 +2208,42 @@ def test_migration(self):
         RBD().migration_commit(ioctx, image_name)
         remove_image()
 
+    def test_migration_with_progress(self):
+        received_callback = False
+        def progress_cb(current, total):
+            nonlocal received_callback
+            received_callback = True
+            return 0
+
+        create_image()
+        RBD().migration_prepare(ioctx, image_name, ioctx, image_name, features=63,
+                                order=23, stripe_unit=1<<23, stripe_count=1,
+                                data_pool=None)
+        RBD().migration_execute(ioctx, image_name, on_progress=progress_cb)
+        eq(True, received_callback)
+        received_callback = False
+
+        RBD().migration_commit(ioctx, image_name, on_progress=progress_cb)
+        eq(True, received_callback)
+        remove_image()
+
     def test_migrate_abort(self):
         create_image()
         RBD().migration_prepare(ioctx, image_name, ioctx, image_name, features=63,
                                 order=23, stripe_unit=1<<23, stripe_count=1,
                                 data_pool=None)
         RBD().migration_abort(ioctx, image_name)
-        remove_image()
```

Comment:
@dillaman Have you removed `remove_image()` intentionally? It seems it is needed (both here and in `test_migrate_abort_with_progress`).

Reply:
Negative, dropped by accident. Thx.